js/gated-content/src/store/modules/personalTraining/webrtc/peerConnection.js

```diff
@@ -0,0 +1,118 @@
+export default {
+  state: {
+    peer: null,
+    peerConnected: false,
+    peerInitializationError: '',
+  },
+  actions: {
+    async initPeer(context) {
+      context.dispatch('debugLog', ['VY Simple Peer', 'initPeer']);
+      // eslint-disable-next-line no-undef
+      if (SimplePeer === undefined) {
+        // eslint-disable-next-line no-undef
+        _.delay(() => {
+          context.dispatch('initPeer');
+        }, 1000);
+        return;
+      }
+
+      let peerjsSTUNUrl;
+      let peerjsTURNUrl;
+      let peerjsTURNUsername;
+      let peerjsTURNCredential;
+
+      ({
+        // eslint-disable-next-line prefer-const
+        peerjsSTUNUrl,
+        // eslint-disable-next-line prefer-const
+        peerjsTURNUrl, peerjsTURNUsername, peerjsTURNCredential,
+        // eslint-disable-next-line prefer-const
+      } = context.getters.getAppSettings);
+
+      const config = {};
+
+      if (peerjsSTUNUrl !== '') {
+        config.iceServers = [
+          { urls: [peerjsSTUNUrl] },
+          {
+            urls: [peerjsTURNUrl],
+            username: peerjsTURNUsername,
+            credential: peerjsTURNCredential,
+          },
+        ];
+      }
+
+      // eslint-disable-next-line no-undef
+      const peer = new SimplePeer({
+        initiator: context.getters.isInstructorRole === true,
+        config,
+      });
+      context.commit('setPeer', peer);
+
+      peer.on('signal', (data) => {
+        context.dispatch('sendSignalingMessage', data);
+      });
+
+      peer.on('connect', () => {
+        context.commit('setPeerConnected', true);
+        context.dispatch('debugLog', ['Peer on connect']);
+        if (context.getters.isJoinedVideoSession) {
+          context.dispatch('callPartner');
+        }
+      });
+
+      peer.on('stream', (stream) => {
+        context.dispatch('debugLog', ['stream received <-', stream]);
+        context.commit('setPeerStreamConnected', true);
+        context.dispatch('setPartnerMediaStream', stream);
+      });
+
+      peer.on('data', (dataString) => {
+        const data = JSON.parse(dataString);
+        context.dispatch('debugLog', ['Peer on data', data]);
```

Comment:
seems like debugLog is for debuggins
Is there any toglle settings to disable/enable it?

Reply:
Yes, this is an additional Vuex store plugin that checks if debug mode was enabled.
https://github.com/fivejars/openy_gated_content/blob/1.4/js/gated-content/src/store/modules/debugLog.js#L4